src/main/java/injectorpractice/service/CarServiceImpl.java

```diff
@@ -0,0 +1,69 @@
+package injectorpractice.service;
+
+import injectorpractice.dao.CarDao;
+import injectorpractice.dao.DriverDao;
+import injectorpractice.lib.Inject;
+import injectorpractice.lib.Service;
+import injectorpractice.model.Car;
+import injectorpractice.model.Driver;
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Service
+public class CarServiceImpl implements CarService {
+    @Inject
+    private CarDao carDao;
+    @Inject
+    private DriverDao driverDao;
+
+    @Override
+    public Car create(Car car) {
+        return carDao.create(car);
+    }
+
+    @Override
+    public Car get(Long id) {
+        return carDao.get(id).get();
+    }
+
+    @Override
+    public List<Car> getAll() {
+        return carDao.getAll();
+    }
+
+    @Override
+    public Car update(Car car) {
+        return carDao.update(car);
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        return carDao.delete(id);
+    }
+
+    @Override
+    public void addDriverToCar(Driver driver, Car car) {
+        if (driverDao.get(driver.getId()).isEmpty()) {
```

Comment:
we don't need this check, when we connect a real db it will handle this situation for us